src/VS1053.cpp

```diff
@@ -427,5 +427,5 @@ void VS1053::loadUserCode(const unsigned short* plugin) {
  * Load the latest generic firmware patch
  */
 void VS1053::loadDefaultVs1053Patches() {
-   loadUserCode(PATCHES);
+   loadUserCode(PATCHES,sizeof(PATCHES)/sizeof(PATCHES[0]));
```

Comment:
Size is already present in the patch code:
https://github.com/baldram/ESP_VS1053_Library/blob/fdbe2a0899cecb39db6bf404f0e49bfd0639311d/src/patches/vs1053b-patches.h#L621

Reply:
Right, those were exactly my thoughts. When using and including exactly one patch, using

```
while (i<PLUGIN_SIZE) 
```

with  ```PLUGIN_SIZE``` defined in the header file and no parameters passed to ```loadUserCode()``` would be the nicest solution.

>Not sure if it is handy to use this when combining patches though.

Agreed, that was the reason for me to pass the plugin size as a parameter. When using multiple plugins, e.g.

```
    if(player.getChipVersion()==4 /*4*/) // MIDI using a VS1053 chip
      {
        player.loadUserCode(MIDI1053, sizeof(MIDI1053)/sizeof(MIDI1053[0])); 
        player.writeRegister(0xA /*SCI_AIADDR*/, 0x50);  // setting VS1053 Start adress for user code               
      }
             
    if(player.getChipVersion()==3) // MIDI using a VS1003 chip
      {      
        player.loadUserCode(MIDI1003, sizeof(MIDI1003)/sizeof(MIDI1003[0]));      
        player.writeRegister(0xA /*SCI_AIADDR*/, 0x30);  // setting VS1003 Start adress for user code                               
      }
```

using the header file defintion of ```PLUGIN_SIZE``` or defining different plugin size names would be at least confusing.

Reply:
Agreed. I don't see a clear and easy (not messy) solution. Maybe @baldram has some idea?

Reply:
Unfortunately no immediate idea. Thank you both for your effort. 

It would be great if we can update a method comment providing a use example, a code snippet. I mean something like [this](https://developer.lsst.io/cpp/api-docs.html#cpp-doxygen-examples).

Then let's merge it. We can always introduce later a single argument method variant.<br>For now it's definitely better to fix the bug.

Reply:
Actually I don't think it's messy, but calling the function with ```PLUGIN_NAME_SIZE``` rather than   ```sizeof(PLUGIN__NAME)/sizeof(PLUGIN__NAME[0])``` might look less complex to the user, I'll change that accordingly and will update the method comment.